YourEasyRent/Controllers/ProductController.cs

```diff
@@ -17,77 +17,139 @@ public class ProductController : ControllerBase
         [HttpGet]
         public async Task<ActionResult<IEnumerable<Product>>> GetProducts()//   не надо бобавлять в скобках Id, потому что мы должне получить все продукты, поэтому и в операторах его нет Task<IEnumerable<Product>> GetProducts()
         {
-            var allProducts = await _repository.GetProducts();
+            try
+            {
+                var allProducts = await _repository.GetProducts();
+                return Ok(allProducts);// получаем все продукты
+            }
 
-            return Ok(allProducts); // получаем все продукты
+            catch (Exception ex) 
+            {
+                Console.WriteLine($"[GetProducts] : {ex.Message}");
+                return StatusCode(422, "Unable to process the request.");
```

Comment:
422 is good, but I think 500 is more appropriate.
Why?
4xx - are fails on the caller side. 
Examples: 
Requests with invalid data = 400 Bad request
Url that does not exist = 404 NotFound

5xx - fails on server side 
Timeout when calling DB = 504 Gateway Timeout
App crash = 500 Internal Server Error